src/components/ListItem.jsx

```diff
@@ -38,7 +46,42 @@ export function ListItem({ name, itemId, dateLastPurchased }) {
 		});
 	};
 
+	//Function to assign string value to buyingUrgency and color value to colorUrgency
+	const getBuyingUrgency = () => {
+		//Returns the difference between the currentDate and dateNextPurchased
+		const daysUntilNextPurchase = getDaysBetweenDates(
+			currentDate,
+			dateNextPurchased.toDate(),
+		);
+
+		//Conditional statement to categorize if an item is:
+		// - inactive: (60 days have passed since the last purchase)
+		// - overdue: currentDate has passed the dateNextPurchased, but not yet inactive
+		// - not soon: (30 days or more until the next purchase)
+		// - kind of soon: (between 7 & 30 days until the next purchase)
+		// - soon: (7 days or fewer until the next purchase)
+		if (daysUntilNextPurchase >= 60) {
```

Comment:
Same question as above about possibly refactoring this to use a `switch` statement

Reply:
This could also be exported as a separate `util` function